### PR TITLE
fix(dashboard,types,js-sdk): Locations & Shipping fixes and cleanup

### PR DESCRIPTION
**Pull request by kasperkristensen**
Still a couple of TODOs in this domain, but the fixes are currently blocked by bugs in the backend. I have added linear issues for those bugs.

Also a bit confused by the intentions behind the pickup and shipping fulfillment sets, see my question in Figma.

RESOLVES CORE-2124, CORE-2125, CORE-2126, CORE-2128, CORE-2129, CORE-2130, CORE-2131, CORE-2123, CORE-2122, CORE-2132, CORE-2133

### CODE REVIEW

File: packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-general-section/location-general-section.tsx
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-general-section/location-general-section.tsx:77]: Currently has the exact same functionality as the Shipping fulfillment set, e.g. you can add return and outbound shipping options, but is this the intended behaviour? Seems to be based on Figma design, but in that case what does it have to do with Pickup?

**Comment by fPolic** [comment]
Looks pretty clean!

I've noticed a few UI things on the first pass:

1. Disabled badge on the list page and details page are different
2. Edit shipping option throws an error now
3. When you enable shipping/pickup and then try disabling it, disabled state is not shown until refresh

**Comment by olivermrbl** [comment]
@fPolic, could I get you to give this another review?

File: packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-general-section/location-general-section.tsx
Comment (olivermrbl) [reply packages/admin-next/dashboard/src/routes/locations/location-detail/components/location-general-section/location-general-section.tsx:77]: Added a comment on this in Figma. This is intended behavior – but perhaps we should consider renaming the concept before the official release. However, I don't think it's super high priority right now. And probably something we could get around just by communicating it correct in admin.

File: packages/admin-next/dashboard/src/routes/locations/location-service-zone-shipping-option-edit/location-service-zone-shipping-option-edit.tsx
Comment (fPolic) [inline packages/admin-next/dashboard/src/routes/locations/location-service-zone-shipping-option-edit/location-service-zone-shipping-option-edit.tsx]: todo: 
```suggestion
    id: so_id
```

File: packages/admin-next/dashboard/src/routes/locations/location-service-zone-shipping-option-edit/location-service-zone-shipping-option-edit.tsx
Comment (fPolic) [inline packages/admin-next/dashboard/src/routes/locations/location-service-zone-shipping-option-edit/location-service-zone-shipping-option-edit.tsx]:
```suggestion
const { location_id, so_id } = useParams()
```

File: packages/admin-next/dashboard/src/hooks/api/fulfillment-sets.tsx
Comment (fPolic) [inline packages/admin-next/dashboard/src/hooks/api/fulfillment-sets.tsx:41]: todo:
```suggestion
  await queryClient.invalidateQueries({
        queryKey: stockLocationsQueryKeys.details(),
      })
```
Add this so when we disable a fulfillment set, the change is reflected in the UI

**Review by fPolic — COMMENTED** [review]
LGTM, just two things to address before merge.

**Review by fPolic — APPROVED** [review]

**Pull request merged**
